core/contracts/contracts/privacy/Shield.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: CC0
+pragma solidity ^0.6.9;
+
+import "./lib/MerkleTree_sha256.sol";
+import "./IShield.sol";
+import "./IVerifier.sol";
+
+contract Shield is IShield, MerkleTree_sha256 {
+    // Observers may wish to listen for nullification of commitments:
+    event NewCommitment(bytes32 newCommitment);
+    event UpdatedCommitment(bytes32 nullifier, bytes32 newCommitment);
```

Comment:
Where are `UpdatedCommitment` and `DeleteCommitment` being used?

If they aren't used yet, I suggest we omit them. It may even be reasonable to just let the merkle tree contract emit its events and just use those instead trying to emit "standard" events here.

Reply:
Removed those events